File: sty_algorithm/src/main/java/com/sty/algorithm/leetcode/editor/cn/[454]四数相加 II.java
package com.sty.algorithm.leetcode.editor.cn;

import java.util.HashMap;
import java.util.Map;

//leetcode submit region begin(Prohibit modification and deletion)
class Solution454 {
    public int fourSumCount(int[] nums1, int[] nums2, int[] nums3, int[] nums4) {
        Map<Integer, Integer> map1 = new HashMap<>();
        for (int n1 : nums1) {
            for (int n2 : nums2) {
                map1.put(n1 + n2, map1.getOrDefault(n1 + n2, 0) + 1);
            }
        }

        int res = 0;
        for (int n1 : nums3) {
            for (int n2 : nums4) {
                res += map1.getOrDefault(-n1 -n2, 0);
            }
        }

        return res;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: sty_algorithm/src/main/scala/com/sty/algorithm/leetcode/editor/cn/[454]四数相加 II.scala
//leetcode submit region begin(Prohibit modification and deletion)
import scala.collection.mutable


object Solution {
    def fourSumCount(nums1: Array[Int], nums2: Array[Int], nums3: Array[Int], nums4: Array[Int]): Int = {
        val map: mutable.HashMap[Int, Int] = mutable.HashMap()
        for (n1 <- nums1) {
            for (n2 <- nums2) {
                map.put(n1 + n2, map.getOrElse(n1 + n2, 0) + 1)
            }
        }

        var res = 0

        for (n1 <- nums3) {
            for (n2 <- nums4) {
                res = res + map.getOrElse(-n1 - n2, 0)
            }
        }

        res
    }
}
//leetcode submit region end(Prohibit modification and deletion)
